### PR TITLE
OCPBUGS-30991: external OIDC: fix certificateAuthority field in structured auth config

### PR DESCRIPTION
**Pull request by sjenning**
replaces #3772

### CODE REVIEW

**Comment by sjenning** [comment]
/override "Red Hat Konflux / hypershift-operator-main-on-pull-request"

File: control-plane-operator/controllers/hostedcontrolplane/kas/auth.go [hunk 1]
```diff
@@ -10,18 +11,23 @@ import (
...
+
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
 	AuthConfigMapKey = "auth.json"
 )
 
-func ReconcileAuthConfig(config *corev1.ConfigMap, ownerRef hcpconfig.OwnerRef, p KubeAPIServerConfigParams) error {
+func ReconcileAuthConfig(config *corev1.ConfigMap, ownerRef hcpconfig.OwnerRef, p KubeAPIServerConfigParams, ctx context.Context, c crclient.Client) error {
```
Comment (deads2k) [inline control-plane-operator/controllers/hostedcontrolplane/kas/auth.go:22]: golang specifies "first arg" for context.

File: control-plane-operator/controllers/hostedcontrolplane/kas/auth.go [hunk 2]
```diff
@@ -39,13 +45,25 @@ func generateAuthConfig(spec *configv1.AuthenticationSpec) *AuthenticationConfig
 		JWT: []JWTAuthenticator{},
 	}
 	if spec == nil {
-		return config
+		return config, nil
 	}
 	for _, provider := range spec.OIDCProviders {
+		caData := ""
+		if provider.Issuer.CertificateAuthority.Name != "" {
+			ca := &corev1.ConfigMap{}
```
Comment (deads2k) [inline control-plane-operator/controllers/hostedcontrolplane/kas/auth.go:53]: will this get triggered when the configmap content is changed?

Comment (sjenning) [reply control-plane-operator/controllers/hostedcontrolplane/kas/auth.go:53]: Yes, HostedControlPlane reconcile is triggered on change of the `Issuer.CertificateAuthority.Name`, changes to the configmap contents, and KAS is redeployed on changes to the auth-config file

**Comment by sjenning** [comment]
/override "Red Hat Konflux / hypershift-operator-dwt4g / hypershift-main-enterprise-contract"

**Pull request merged**